pkg/ingester/ingester_v2_test.go

```diff
@@ -1815,6 +1815,48 @@ func TestIngester_flushing(t *testing.T) {
 				m.AssertNumberOfCalls(t, "Sync", 1)
 			},
 		},
+
+		"noBlockSpanningOverADayOnFlush": {
+			setupIngester: func(cfg *Config) {
+				cfg.BlocksStorageConfig.TSDB.FlushBlocksOnShutdown = false
+			},
+
+			action: func(t *testing.T, i *Ingester, m *shipperMock) {
+				// Pushing 5 samples, spanning over 3 days.
```

Comment:
> spanning over 3 days

Aren't they over 2 days? However I believe the test would be even stronger if samples would span 3 days and you would expect 3 blocks in output.

Reply:
> However I believe the test would be even stronger if samples would span 3 days and you would expect 3 blocks in output.

Reply:
> However I believe the test would be even stronger if samples would span 3 days and you would expect 3 blocks in output.

It will depend on the actual samples, because by default it will try to create 2h blocks and not over a day. So a test of creating smaller blocks + breaking the block when it spans 2 day might be fine. I will update the test and maybe have 3 days itself anyway.